bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/util/time/Interval.java

```diff
@@ -38,16 +38,21 @@ public class Interval implements TemporalAmount {
   private final Period period;
   private final Duration duration;
   private final Optional<ZonedDateTime> start;
+  private final boolean firstExecution;
```

Comment:
Introducing a new field is an option to fix the issue :+1: 

But we could avoid the new field by checking how the interval is used. Initially, the interval is created via the constructor (i.e. parsing). When the timer is triggered, then we use `.withStart()` to create a new interval: https://github.com/camunda/zeebe/blob/61c3d2013d366826b5c56318adc822f2dbfc9c20/engine/src/main/java/io/camunda/zeebe/engine/processing/timer/TriggerTimerProcessor.java#L168

:x: If a start time is set then we could always return the start time on calling `.toEpochMilli()`. When a new interval is created via `.withStart()` then we could add the duration to the given time. So, the start time is the due date of the timer.

---

:x: We should add a new test case to `bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/util/time/RepeatingIntervalTest.java` to verify how the due dates are calculated. First, `.toEpochMilli()` returns the start time. After calling `.withStart()` with the last due date, `.toEpochMilli()` returns the last due date + duration.